### ISSUE TITLE
Use LLaMA.cpp through library instead of standalone exectuable

### ISSUE THREAD

**Issue by gudrob** [gudatr/godot-ai-rpg#4]
The standalone exectuable brings permission problems with it and also the http server and slot management adds unnecessary overhead. The Project will migrate to using llama through a library in the future.

**Comment by gudrob** [comment]
Apple Silicon is migrated, windows support is currently therefore broken until also migrated

**Comment by gudrob** [comment]
Windows has also bin migrated but needs some more testing as it affects the MacOS implementation aswell

**Comment by Koalamana9** [comment]
How to compile custom llama-server.dll? Is there any changes in llama.cpp codebase or any recent version will work?

**Comment by gudrob** [comment]
Hey @Koalamana9,

it is this repo: https://github.com/gudrob/llama.cpp
Then just like you build regular llama cpp and a server lib will be created.

Unfortunately the approach I took there was not sustainable and broke like 5 commits later as they overhauled sampling and I tbh had no idea how to merge that with the changes I made. So anything past Llama 3.1 wont work 🎻

I was thinking of migrating the LLM to Llama Sharp and the TTS to Kokoro on Sherpa Onnx but I cant tell you when exactly that will happen, sorry.
